src/App.js

```diff
@@ -1,42 +1,76 @@
 //imports
-import React from 'react';
+import React, { Component } from 'react';
 import Header from './Header/Header.js';
 import FilmCollection from './FilmCollection/FilmCollection';
 import Footer from './Footer/Footer.js';
 import ResultsArea from './ResultsArea/ResultsArea';
 import ErrorBoundary from './ErrorBoundary/ErrorBoundary';
+import { fetchRequestIfNeeded } from './redux-utils/asyncActionUtils';
 import { connect } from 'react-redux';
+import { Route, Switch } from 'react-router-dom';
 import './style.css';
 
-const App = (props) => {
-
-    const { films, isFilmClicked, genre, filmsSameGenre } = props;
-    const filmsNumber = props.films.length;
-
-    return (
-        <div className='app'>
-            <ErrorBoundary>
-                <Header isFilmClicked={isFilmClicked} />
-            </ErrorBoundary>
-            <ErrorBoundary>
-                <ResultsArea isFilmClicked={isFilmClicked} genre={genre} filmsNumber={filmsNumber} />
-            </ErrorBoundary>
-            <ErrorBoundary>
-                <FilmCollection filmsNumber={filmsNumber} films={films} filmsSameGenre={filmsSameGenre} isFilmClicked={isFilmClicked} />
-            </ErrorBoundary>
-            <Footer/>
-        </div>
-    );
-};
+class App extends Component {
+
+    constructor(props) {
+        super(props);
+        this.props = props;
+    }
+
+    componentDidMount() {
+        const id = this.props.match.params.id;
```

Comment:
I don't think that it's a good idea to fetch data on the highest level. How about moving it in component that renders the list of films?